src/util/usertools.c

```diff
@@ -587,26 +601,26 @@ errno_t sss_user_by_name_or_uid(const char *input, uid_t *_uid, gid_t *_gid)
             return ret;
         }
 
-        /* Nope, maybe a username? */
-        pwd = getpwnam(input);
+        status = nss_ops.getpwnam_r(input, &pwd, s_nss_buffer, NSS_BUFFER_SIZE, &errnop);
     } else {
-        pwd = getpwuid(uid);
+        status = nss_ops.getpwuid_r(uid, &pwd, s_nss_buffer, NSS_BUFFER_SIZE, &errnop);
     }
 
-    if (pwd == NULL) {
+    if (status != NSS_STATUS_SUCCESS) {
         DEBUG(SSSDBG_OP_FAILURE,
               "[%s] is neither a valid UID nor a user name which could be "
               "resolved by getpwnam().\n", input);
```

Comment:
Log both `status` and `errnop`.

Reply:
Can you take a look at the log message and tell me if it makes sense?

Reply:
Yes, thanks.